web/src/portico/signup.ts

```diff
@@ -342,3 +343,102 @@ $(() => {
         }
     });
 });
+
+function get_or_set(
+    input_selector: string,
+    keep_leading_whitespace?: boolean,
+    no_trim?: boolean,
+): (newval?: string) => string {
+    return function () {
+        const $elem = $<HTMLInputElement | HTMLTextAreaElement>(input_selector);
+
+        if (!$elem.length) {
+            return "";
+        }
+
+        const oldval = $elem.val();
+        if (oldval === undefined) {
+            return "";
+        }
+
+        if (no_trim) {
+            return oldval;
+        } else if (keep_leading_whitespace) {
+            return oldval.trimEnd();
+        }
+        return oldval.trim();
+    };
+}
+
+let password_too_long = false;
+
+function set_password_too_long(status: boolean, is_signup: boolean): void {
+    password_too_long = status;
+    if (is_signup) {
+        update_sign_up_button_status();
+    } else {
+        update_reset_password_button_status();
+    }
+}
+
+function update_sign_up_button_status(): void {
+    $("#signup-button")
+        .toggleClass("disabled-signup-button", password_too_long)
```

Comment:
I think there is no use of this class and the CSS can be added based on the `disabled` attribute itself.